### PR TITLE
Feat: Support sessions in the BigQuery adapter

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/core/engine_adapter/bigquery.py [hunk 1]
```diff
@@ -87,6 +87,18 @@ def _job_params(self) -> t.Dict[str, t.Any]:
             params["maximum_bytes_billed"] = self._extra_config.get("maximum_bytes_billed")
         return params
 
+    def _begin_session(self) -> None:
+        from google.cloud.bigquery import QueryJobConfig
+
+        job = self.client.query("SELECT 1;", job_config=QueryJobConfig(create_session=True))
```
Comment (eakmanrq) [inline sqlmesh/core/engine_adapter/bigquery.py:93]: Ah so you can't actually just pass in a session ID then? I was thinking you had to get it from BQ.

File: tests/core/engine_adapter/test_bigquery.py [hunk 2]
```diff
@@ -364,3 +364,18 @@ def test_merge(mocker: MockerFixture):
         "job_config",
         "table",
     ]
+
+
+def test_begin_end_session(mocker: MockerFixture):
```
Comment (eakmanrq) [inline tests/core/engine_adapter/test_bigquery.py:369]: Does it also make sense to ensure that a series of engine adapter calls within a single session share a common session ID on their jobs?

File: sqlmesh/core/engine_adapter/bigquery.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/engine_adapter/bigquery.py:93]: yep, you were right after all.

File: tests/core/engine_adapter/test_bigquery.py (hunk 2 above)
Comment (izeigerman) [reply tests/core/engine_adapter/test_bigquery.py:369]: Updated the test

File: sqlmesh/core/engine_adapter/bigquery.py (hunk 1 above)
Comment (eakmanrq) [reply sqlmesh/core/engine_adapter/bigquery.py:93]: Cool making sure we still explored passing it in. That would have been nice. 😃

**Review by eakmanrq — APPROVED** [review]

**Pull request merged**